libs/map/depth_point_map.cpp

```
@@ -0,0 +1,150 @@
+size_t DepthPointMap::count_close_and_visible(const DepthCameraInfo& cam) const {
+  const float max_depth = settings_.visible_max_depth;
+  size_t count = 0;
+  for (const auto& p : points_) {
+    if (close_and_visible(p, cam, max_depth)) {
+      ++count;
+    }
+  }
+  return count;
+}
+
+void DepthPointMap::add_samples(const DepthCameraInfo& cam, size_t n) {
```

Clang (14.0.6)

[warning] method 'add_samples' can be made static

(readability-convert-member-functions-to-static)